Format validation; redo of PR 3116

benjiwheeler opened the pull request:
Retry of https://github.com/LLK/scratch-www/pull/3116, after it was reverted in https://github.com/LLK/scratch-www/pull/3125

### Resolves:

Resolves https://github.com/LLK/scratch-www/issues/3115

### Changes:

* Adds `validationClassName` parameter to `FormikInput` component
* Set formatting of validations in join flow using `validationClassName`

Note that unlike in https://github.com/LLK/scratch-www/pull/3116, I'm _not_ changing the css I use to prevent margins from collapsing. After this PR, I will consider ways to prevent margin collapse without interfering with other modals' formatting.

rschamp commented on src/components/formik-forms/formik-input.jsx:
```diff
@@ -11,9 +11,10 @@ require('../forms/row.scss');
     ...props
 }) => (
-    <div className="col-sm-9 row">
+    <div className="col-sm-9 row row-relative">
```
What requires the row to be relative? Is it because it might have a validation message? Right now it's not clear to me from this why I would need this class or not if I were extending this or re-using it in the future. Can the class name refer to why it's needed (e.g. `has-validation` or something specific like that) so that's self-documenting?

benjiwheeler replied:
Renamed -- LMK what you think.

rschamp commented:
Checking it out on scratch.ly it looks like there are some regressions in the styles related to this.

**Production**
![image](https://user-images.githubusercontent.com/1512021/61225730-2afb0800-a6ef-11e9-9448-079f8d29e376.png)

**Staging**
![image](https://user-images.githubusercontent.com/1512021/61225762-3a7a5100-a6ef-11e9-8b2e-b291bd2ed91f.png)

benjiwheeler commented:
The overflow/clipping issue was from another PR; a fix for that is at https://github.com/LLK/scratch-www/pull/3140. I'll update this PR after that one goes in.

rschamp commented:
Sounds good — I still think that class name could be improved since it's not very semantic.

benjiwheeler commented:
> Sounds good — I still think that class name could be improved since it's not very semantic.

Yes, I'm thinking of `.row-with-tooltip` since the absolutely positioned notice can be a validation error or just a friendly FYI. What do you think?

rschamp commented:
Sounds good to me!

benjiwheeler commented:
OK, updated and ready for review again. Updated look: no longer clipping.

![image](https://user-images.githubusercontent.com/3431616/61325101-fcaa2500-a7e1-11e9-9836-cf36003bcb56.png)

![image](https://user-images.githubusercontent.com/3431616/61325104-fddb5200-a7e1-11e9-97d4-2fa32d091a1b.png)
